### PR TITLE
*: refine executionSummary

### PR DESCRIPTION
**Pull request by ywqzzy**
### What problem does this PR solve?

Issue Number: ref #5900 

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Flash/Coprocessor/ExecutionSummary.h [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    UInt64 time_processed_ns;
+    UInt64 num_produced_rows;
+    UInt64 num_iterations;
+    UInt64 concurrency;
+    ExecutionSummary()
+        : time_processed_ns(0)
+        , num_produced_rows(0)
+        , num_iterations(0)
+        , concurrency(0)
+    {}
```
Comment (fuzhe1989) [inline dbms/src/Flash/Coprocessor/ExecutionSummary.h:33]:
```suggestion
    UInt64 time_processed_ns = 0;
    UInt64 num_produced_rows = 0;
    UInt64 num_iterations = 0;
    UInt64 concurrency = 0;
    ExecutionSummary() = default;
```

File: dbms/src/Flash/Coprocessor/ExecutionSummaryCollector.h [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+namespace DB
+{
+class ExecutionSummaryCollector
+{
+public:
+    explicit ExecutionSummaryCollector(
+        DAGContext & dag_context_)
+        : dag_context(dag_context_)
+    {
+        for (auto & p : dag_context.getProfileStreamsMap())
```
Comment (ywqzzy) [inline dbms/src/Flash/Coprocessor/ExecutionSummaryCollector.h:29]: Not exception safe.

File: dbms/src/Flash/Coprocessor/ExecutionSummaryCollector.h [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        }
+    }
+
+    void fillTiExecutionSummary(
+        tipb::ExecutorExecutionSummary * execution_summary,
+        ExecutionSummary & current,
+        const String & executor_id,
+        bool delta_mode);
+
+    void addExecuteSummaries(tipb::SelectResponse & response, bool delta_mode);
```
Comment (SeaRise) [inline dbms/src/Flash/Coprocessor/ExecutionSummaryCollector.h:41]:
```suggestion
    void addExecuteSummaries(tipb::SelectResponse & response, bool delta_mode);

private:
    void fillTiExecutionSummary(
        tipb::ExecutorExecutionSummary * execution_summary,
        ExecutionSummary & current,
        const String & executor_id,
        bool delta_mode);
```

**Review by SeaRise — APPROVED** [review]
LGTM

**Pull request merged**